### PR TITLE
feat: add support for clone

### PR DESCRIPTION
**Pull request by Neenu1995**

### CODE REVIEW

**Comment by Neenu1995** [comment]
@shollyman PTAL.

File: google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java [hunk 1]
```diff
@@ -5179,4 +5181,58 @@ public void testCreateExternalTableWithReferenceFileSchemaParquet() {
...
+    CopyJobConfiguration createdConfiguration = createdJob.getConfiguration();
+    assertNotNull(createdConfiguration.getSourceTables());
+    assertNotNull(createdConfiguration.getOperationType());
+    assertNotNull(createdConfiguration.getDestinationTable());
+    Job completedJob = createdJob.waitFor();
+    assertNull(completedJob.getStatus().getError());
+
+    Table cloneTable = bigquery.getTable(DATASET, cloneTableName);
+    assertNotNull(cloneTable);
+    System.out.println(cloneTable.getDefinition().getType());
```
Comment (shollyman) [inline google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java:5223]: Guessing the System.out.println lines in here are stale debug statements that survived?

**Review by shollyman — APPROVED** [review]
Thanks for adding this!  I'm always a little hesistant with extensions to TableDefinition due to it being a java-only pattern.  I wonder if we should have a generic mechanism somewhere for populating the known common fields within the various table definition subclasses.

File: google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java (hunk 1 above)
Comment (Neenu1995) [reply google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java:5223]: Yes

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TableDefinition.java [hunk 2]
```diff
@@ -85,6 +85,8 @@ public Type apply(String constant) {
 
     public static final Type SNAPSHOT = type.createAndRegister("SNAPSHOT");
 
+    public static final Type CLONE = type.createAndRegister("CLONE");
```
Comment (shollyman) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TableDefinition.java:88]: Oh, here's the issue.  We likely shouldn't make a new CloneTableDefinition at all.  The type returned by the table metadata for a clone tables is still TABLE, not CLONE.  It just has the CloneTableDefinition present.

Comment (Neenu1995) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TableDefinition.java:88]: Got it.. That Makes a lot more sense. By that definition shouldn't SNAPSHOT also follow a similar implementation flow? Or is that different?
Does creating a SNAPSHOT table change its definition?

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TableInfo.java [hunk 3]
```diff
@@ -205,6 +211,7 @@ static class BuilderImpl extends Builder {
       this.labels = Labels.fromPb(tablePb.getLabels());
       this.requirePartitionFilter = tablePb.getRequirePartitionFilter();
       this.defaultCollation = tablePb.getDefaultCollation();
+      this.cloneDefinition = CloneDefinition.fromPb(tablePb.getCloneDefinition());
```
Comment (shollyman) [inline google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TableInfo.java:214]: Not sure if you want it here or elsewhere in fromPb, but don't you need to handle null cloneDefinition?

Comment (Neenu1995) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/TableInfo.java]: Fixed

**Pull request merged**